modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -25,7 +26,7 @@ package object engine {
     * This represents an actual real-world action to be done in the underlying
     * systems.
     */
-  type Action = Task[Result]
+  type Action = Kleisli[Task, ActionMetadata, Result]
```

Comment:
Unconstrained use of transformers? That's reading too much into this 😄 
I'd rather use other techniques as we have seen that transformers make the code very hard to follow